src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,48 @@
+package duke.task;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a task with a time to do it by. A <code>Deadline</code> object corresponds to the task represented by
+ * a String name and a String by which specified the date to do it by.
+ */
+public class Deadline extends duke.task.Task {
+    protected String by;
+
+    /**
+     * Returns a new instance of a <code>Deadline</code> object with the specified name and date to do it by.
+     * @param name Name of the deadline task.
+     * @param by Date to do the deadline task by.
+     */
+    public Deadline(String name, String by) {
+        super(name);
+        System.out.println(by);
+        try {
+            LocalDate d1 = LocalDate.parse(by);
+            this.by = d1.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
```

Comment:
maybe I didn't understand your code properly but I'm not very sure why the date format is MMM d yyyy HAHA